DQM/SiStripCommon/src/TkHistoMap.cc

```diff
@@ -78,27 +85,43 @@ void TkHistoMap::loadTkHistoMap(std::string path, std::string MapName, bool mech
 }
 
 void TkHistoMap::createTkHistoMap(std::string& path, std::string& MapName, float& baseline, bool mechanicalView){
-  
+
   int nchX;
   int nchY;
   double lowX,highX;
   double lowY, highY;
   std::string fullName, folder;
 
-  tkHistoMap_.resize(HistoNumber);    
+  const bool bookTH2F = isTH2F_;
+  tkHistoMap_.resize(HistoNumber);
   for(int layer=1;layer<HistoNumber;++layer){
     folder=folderDefinition(path,MapName,layer,mechanicalView,fullName);
     tkdetmap_->getComponents(layer,nchX,lowX,highX,nchY,lowY,highY);
-    MonitorElement* me  = dqmStore_->bookProfile2D(fullName.c_str(),fullName.c_str(),
+    MonitorElement* me;
+
+    if(bookTH2F==false){
+
+    me  = dqmStore_->bookProfile2D(fullName.c_str(),fullName.c_str(),
```

Comment:
Understood, what I don't understand is why we discuss this problem in this PR. This method was already in the code.
 
I **assume** that this part of the code is not called in the [DQMOffline] (https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/DQMOffline/Configuration/python/DQMOffline_cff.py).
I am running a custom test to check if this is the case.